.eslintrc

```diff
@@ -25,221 +31,145 @@
-    "one-var": [0, "never"],
-    "padded-blocks": [0, "never"],
-    "quote-props": [0, "as-needed"],
-    "quotes": [0, "single"],
-    "semi-spacing": [
-      1,
-      {
-        "before": false,
-        "after": true
-      }
-    ],
-    "semi": [1, "always"],
-    "sort-vars": 0,
-    "space-after-keywords": 0,
-    "space-before-blocks": [0, "always"],
-    "space-before-function-paren": [0, "never"],
-    "space-in-parens": [0, "never"],
-    "space-infix-ops": 0,
-    "space-unary-ops": 0,
-    "spaced-comment": [0, "always"],
-    "wrap-regex": 0,
+    "no-unneeded-ternary": 1 /* ECMAScript 6 */,
+    "prefer-const": "error"
+  },
+  "overrides": [
+    {
+      "files": ["**/*.ts", "**/*.tsx"],
+      "extends": [
+        "eslint:recommended",
+        "plugin:react/recommended",
+        "plugin:react-hooks/recommended",
+        "plugin:@typescript-eslint/recommended"
+      ],
+      "parser": "@typescript-eslint/parser",
+      "parserOptions": {
+        "ecmaFeatures": {
+          "jsx": true,
+          "modules": true,
+          "legacyDecorators": true
+        },
+        "ecmaVersion": 2018,
+        "sourceType": "module",
+        "project": "./tsconfig.json"
+      },
+      "plugins": ["react", "@typescript-eslint"],
+      "rules": {
+        "react/jsx-boolean-value": ["error", "never", { "always": [] }],
+        "react/jsx-closing-bracket-location": ["error", "line-aligned"],
+        "react/jsx-closing-tag-location": "error",
+        "react/jsx-curly-spacing": [
+          "error",
+          "never",
+          { "allowMultiline": true }
+        ],
+        "react/no-arrow-function-lifecycle": "error",
+        "react/no-invalid-html-attribute": "error",
+        "react/jsx-no-useless-fragment": "error",
+        "react/jsx-no-constructed-context-values": "error",
+        "react/jsx-fragments": ["error", "syntax"],
+        "react/jsx-no-duplicate-props": ["error", { "ignoreCase": true }],
+        "react/jsx-pascal-case": [
+          "error",
+          {
+            "allowAllCaps": true,
+            "ignore": []
+          }
+        ],
+        "react/no-danger": "warn",
+        "react/no-did-update-set-state": "error",
+        "react/no-will-update-set-state": "error",
+        "react/self-closing-comp": "error",
+        "react/jsx-no-undef": ["error", { "allowGlobals": true }],
+        "react/prop-types": "off",
+
+        "@typescript-eslint/no-non-null-assertion": "off",
+        "@typescript-eslint/no-explicit-any": "off",
+        "@typescript-eslint/no-namespace": "off",
+        "@typescript-eslint/no-var-requires": "off",
+        "@typescript-eslint/no-this-alias": "off",
 
-    /* ECMAScript 6 */
-    "constructor-super": 1,
-    "generator-star-spacing": 0, // see Babel section
-    "no-this-before-super": 1,
-    "no-var": 0,
-    "object-shorthand": 0, // see Babel section
-    "prefer-const": 1,
-    "no-useless-escape": 0
-  }
+        // @TODO: enable these rules
```

Comment:
May I suggest putting this as `warn` instead of turning them `off`, and ditching the rule of max-warnings 0 when running linter. Having them as `warn` will allow developers to fix them incrementally during development, if the rules are turned off then we can't get any meaningful information that those even exist during development. 
I agree that treating warnings as errors in linter might be beneficial so more rules can be enforced, but for me, it is pretty much the same as having it error by default. 

What do you think about this? @ljowen @nf-s

Reply:
I think that's a good idea, I'd like to resolve these as a follow up PR (and change these back to the default "error"). So I've set --warn-limit=341 in gulpfile. To me it doesn't make sense to have warnings=0, since that just means warnings become errors.